app/src/main/java/com/mapbox/services/android/navigation/testapp/example/ui/ExampleViewModel.kt

```diff
@@ -0,0 +1,143 @@
+package com.mapbox.services.android.navigation.testapp.example.ui
+
+import android.app.Application
+import android.arch.lifecycle.AndroidViewModel
+import android.arch.lifecycle.MutableLiveData
+import android.location.Location
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEnginePriority
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.geocoding.v5.GeocodingCriteria
+import com.mapbox.api.geocoding.v5.MapboxGeocoding
+import com.mapbox.api.geocoding.v5.models.GeocodingResponse
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.services.android.navigation.testapp.NavigationApplication.Companion.instance
+import com.mapbox.services.android.navigation.testapp.R
+import com.mapbox.services.android.navigation.testapp.example.ui.navigation.*
+import com.mapbox.services.android.navigation.ui.v5.voice.NavigationSpeechPlayer
+import com.mapbox.services.android.navigation.ui.v5.voice.SpeechPlayerProvider
+import com.mapbox.services.android.navigation.v5.milestone.Milestone
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import timber.log.Timber
+import java.util.Locale.US
+
+class ExampleViewModel(application: Application) : AndroidViewModel(application) {
+
+  companion object {
+    const val ONE_SECOND_INTERVAL = 1000
+  }
+
+  val location: MutableLiveData<Location> = MutableLiveData()
```

Comment:
We should explore the option of using custom `LiveData`s and `postValue` instead. Right now, because we're exposing `MutableLiveData` any other outside class could change mistakenly the values with `setValue`. The same thing applies to the rest of variables 👇

Reply:
@Guardiola31337 so like a custom class that extends from `MutableLiveData` and only exposes the `postValue` method?

Reply:
`MutableLiveData` exposes [`postValue`](https://developer.android.com/reference/android/arch/lifecycle/MutableLiveData.html#postValue(T)). I was referring to using that one instead.